evm/src/witness/util.rs

```diff
@@ -135,9 +135,18 @@ pub(crate) fn mem_write_gp_log_and_fill<F: Field>(
     channel.addr_context = F::from_canonical_usize(address.context);
     channel.addr_segment = F::from_canonical_usize(address.segment);
     channel.addr_virtual = F::from_canonical_usize(address.virt);
-    for (i, limb) in val_limbs.into_iter().enumerate() {
-        channel.value[2 * i] = F::from_canonical_u32(limb as u32);
-        channel.value[2 * i + 1] = F::from_canonical_u32((limb >> 32) as u32);
+    if row.is_keccak_sponge.is_one() {
```

Comment:
Great to see this fixed! I was thinking about changing the endianness of our Keccak STARK, to match EVM which IIUC maps Keccak's byte output to a U256 based on BE ordering. This seems a lot easier though.

Would it be easy to modify the byte order in `generate_keccak_general` instead? That would feel most natural to me since this feels pretty specific to Keccak. But if not I think this is good for now.

Reply:
I can have a look!

Reply:
I've updated it to perform the change in `generate_keccak_general` as you suggested.